lua/plenary/path.lua

```diff
@@ -515,30 +515,62 @@ function Path:rmdir()
   uv.fs_rmdir(self:absolute())
 end
 
+---Rename this file or directory to the provided path (`opts.new_name`),
+---returning a new Path instance upon success.  The rename is aborted if the
+---new path already exists. Relative paths are interpreted relative to the
+---current working directory.
+---@generic T: Path
```

Comment:
Is this a string or plenary.Path (table of strings) ? Consider declaring a Path class, if you think this would be useful.